File: AtCoder/Greens/cpp/ABC019-C-Takahashi_kun_to_Mahou_no_Hako.cpp
#include <bits/stdc++.h>
#define MAX_N 100'000
using namespace std;
typedef long long ll;

ll N;

int main() {
  cin >> N;
  set<ll> st;
  for (ll i = 0; i < N; ++i) {
    ll A;
    cin >> A;
    st.insert(A);
  }
  ll res = 0;
  for (auto s : st) {
    if (s % 2 != 0) {
      ++res;
      continue;
    }
    bool f = true;
    while (s % 2 == 0) {
      s /= 2;
      if (st.count(s)) {
        f = false;
        break;
      }
    }
    if (f) ++res;
  }
  printf("%lld\n", res);
}

File: AtCoder/Greens/python/ABC019-C-Takahashi_kun_to_Mahou_no_Hako.py
N = int(input())
A = set(map(int, input().split()))
res = 0
for a in A:
    if a % 2 != 0:
        res += 1
        continue
    f = True
    while a % 2 == 0:
        a //= 2
        if a in A:
            f = False
            break
    if f:
        res += 1
print(res)
